### ISSUE TITLE
OSError with chess.py

### ISSUE THREAD

**Issue by Yoiro** [ReactiveX/RxPY#267]
Hi, 

While searching for resources about merging RxPy with PyGame, I just found your chess.py example.

I copied your code and wanted to check if it ran OK, but when I do my `python chess.py`, I run into an OSError (see full log below). I'm using Windows 10 (with Git Bash) and Python 3.6.2. Also I tried to add some logging inside your SchedulerBase.to_datetime() in order to understand this error a bit more, but without much result.

I'd be glad to help solve this one but I can't see where the error can come from so if anybody here can help me find a way that would be very much appreciated.

Thanks.

```
$ pip freeze
pygame==1.9.4
Rx==1.6.1

$ python chess.py
pygame 1.9.4
Hello from the pygame community. https://www.pygame.org/contribute.html
<class 'int'> 639 # ------ logger.error("%s %s", % (type(duetime), duetime)
0.639 # ------------------ logger.error(str(duetime/1000.0))
Traceback (most recent call last):
  File "chess.py", line 91, in <module>
    main()
  File "chess.py", line 69, in main
    mousemove.on_next(pos)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\subjects\subject.py", line 92, in on_next
    observer.on_next(value)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\core\observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\core\autodetachobserver.py", line 16, in _on_next_core
    self.observer.on_next(value)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\core\observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\core\anonymousobserver.py", line 14, in _on_next_core
    self._next(value)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\linq\observable\materialize.py", line 19, in on_next
    observer.on_next(OnNext(value))
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\core\observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\core\autodetachobserver.py", line 16, in _on_next_core
    self.observer.on_next(value)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\core\observerbase.py", line 19, in on_next
    self._on_next_core(value)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\core\anonymousobserver.py", line 14, in _on_next_core
    self._next(value)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\linq\observable\select.py", line 33, in on_next
    observer.on_error(err)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\core\observerbase.py", line 33, in on_error
    self._on_error_core(error)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\core\autodetachobserver.py", line 23, in _on_error_core
    self.observer.on_error(exn)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\core\observerbase.py", line 33, in on_error
    self._on_error_core(error)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\core\anonymousobserver.py", line 17, in _on_error_core
    self._error(error)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\internal\basic.py", line 33, in default_error
    raise err
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\linq\observable\select.py", line 31, in on_next
    result = selector(value, count[0])
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\internal\utils.py", line 57, in func_wrapped
    ret = fn(*args, **kw)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\internal\utils.py", line 46, in func1
    return func(arg1)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\linq\observable\timestamp.py", line 28, in selector
    return Timestamp(value=x, timestamp=scheduler.now)
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\concurrency\mainloopscheduler\pygamescheduler.py", line 79, in now
    return self.to_datetime(pygame.time.get_ticks())
  File "C:\dev\prj\intellichess\.venv\lib\site-packages\rx\concurrency\schedulerbase.py", line 83, in to_datetime
    duetime = datetime.fromtimestamp(duetime/1000.0)
OSError: [Errno 22] Invalid argument

**Comment by jcafhe** [comment]
Hi,

I have tried the chess example with Rx 1.6.1, pygame 1.9.4, python 3.7 and 2.7, on archlinux and it works for me.

Could you provide more details about your setup (os, python version, ...) ?

**Comment by jcafhe** [comment]
I have tried to run the example on win7 with python 3.7.0 and it crashs with the same error, but not with python 3.7.2.

It seems to be related with the underliying C library and the localtime function on windows os. Therefore, in python, datetime.datetime.fromtimestamp() raises OSError if the timestamp is  lower than 86400.0.
https://bugs.python.org/issue29097

According to the [changelog](https://docs.python.org/3/whatsnew/changelog.html#id27) (bpo-29097), this is fixed in python **3.7.1** so you should update your version.

**Comment by dbrattli** [comment]
Fixes added for master branch as well. Closing this issue.
